### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by shyamsgru**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
-# expert-ui-framework
\ No newline at end of file
+# Autocomplete Component Application
```
Comment (ram-zhapix) [inline README.md:1]: Good to see README.md

But this file is common across framework including all components

File: package.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
+  "name": "autocomplete-component",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/user-event": "^13.5.0",
```
Comment (ram-zhapix) [inline package.json:6]: @testing-library/user-event -  Used during development and not useful for production code. 

Hence move to dev dependencies

File: package.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "@types/jest": "^27.5.2",
+    "@types/node": "^16.18.105",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
```
Comment (ram-zhapix) [inline package.json:10]: move @types to dev dependencies

File: src/component/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  suggestions: string[];
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = ({ suggestions }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:13]: check for null values  at second level. 

you may use like e.currentTarget?.value

File: src/component/Autocomplete.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+export interface AutocompleteProps {
+  suggestions: string[];
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = ({ suggestions }) => {
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:7]: good to see destructuring..

File: src/component/Autocomplete.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  const [inputValue, setInputValue] = useState('');
+  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;
+    setInputValue(userInput);
+
+    const filtered = suggestions.filter(
+      suggestion => suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:17]: check for null values of 'suggestion'

File: src/component/Autocomplete.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    );
+    setFilteredSuggestions(filtered);
+    setShowSuggestions(true);
+  };
+
+  const handleClick = (suggestion: string) => {
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:23]: try to use the hook useCallback for event handlers

File: src/logo.svg
Comment (ram-zhapix) [inline src/logo.svg:1]: Let us remove this file

File: src/component/Autocomplete.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    setInputValue(suggestion);
+    setFilteredSuggestions([]);
+    setShowSuggestions(false);
+  };
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:27]: Refer this document while handling batch updates for multiple state values

https://react.dev/learn/queueing-a-series-of-state-updates

File: src/component/Autocomplete.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        type="text"
+        className="form-control"
+        value={inputValue}
+        onChange={handleChange}
+      />
+      {showSuggestions && inputValue && (
+        <ul className="list-group">
+          {filteredSuggestions.map((suggestion, index) => (
+            <li
+              key={index}
```
Comment (ram-zhapix) [inline src/component/Autocomplete.tsx:41]: good

File: src/reportWebVitals.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
```
Comment (ram-zhapix) [inline src/reportWebVitals.ts:3]: try to understand the importance of this file

File: src/component/Autocomplete.test.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Autocomplete from './Autocomplete';
+
+test('renders Autocomplete component', () => {
+  render(<Autocomplete suggestions={['Apple', 'Banana', 'Cherry']} />);
+  const inputElement = screen.getByRole('textbox');
```
Comment (ram-zhapix) [inline src/component/Autocomplete.test.tsx:7]: getByRole is good approach

**Review by ram-l-coe — CHANGES_REQUESTED** [review]
Please fix the review comments

**Comment by ram-l-coe** [comment]
@shyamsgru - Please fix review comments and resolve conflicts

File: src/component/Autocomplete.tsx (hunk 8 above)
Comment (shyamsgru) [reply src/component/Autocomplete.tsx:27]: kk sir

File: src/reportWebVitals.ts (hunk 10 above)
Comment (shyamsgru) [reply src/reportWebVitals.ts:3]: kk sir

File: src/component/Autocomplete.test.tsx (hunk 11 above)
Comment (shyamsgru) [reply src/component/Autocomplete.test.tsx:7]: thank you sir

**Pull request closed without merging**